File: 30daysOFcode_Go/solution_set2/day 02.go
package main

import (
	"bufio"
	"fmt"
	"io"
	"math"
	"os"
	"strconv"
	"strings"
)

// remember to import "math"

// Complete the solve function below.
func solve(meal_cost float64, tip_percent int32, tax_percent int32) {
	// calculating total according to given condition
	tip := float64(tip_percent) / 100 * meal_cost
	tax := float64(tax_percent) / 100 * meal_cost
	total := tip + tax + meal_cost
	total = math.Round(total)
	fmt.Println(total)
}

func main() {
	reader := bufio.NewReaderSize(os.Stdin, 1024*1024)

	meal_cost, err := strconv.ParseFloat(readLine(reader), 64)
	checkError(err)

	tip_percentTemp, err := strconv.ParseInt(readLine(reader), 10, 64)
	checkError(err)
	tip_percent := int32(tip_percentTemp)

	tax_percentTemp, err := strconv.ParseInt(readLine(reader), 10, 64)
	checkError(err)
	tax_percent := int32(tax_percentTemp)

	solve(meal_cost, tip_percent, tax_percent)
}

func readLine(reader *bufio.Reader) string {
	str, _, err := reader.ReadLine()
	if err == io.EOF {
		return ""
	}

	return strings.TrimRight(string(str), "\r\n")
}

func checkError(err error) {
	if err != nil {
		panic(err)
	}
}

File: 30daysOFcode_Dart/day 02.dart
import 'dart:io';

void main() {
  double meal_cost;
  int tipPercent, taxPercent;

  meal_cost = double.parse(stdin.readLineSync());
  tipPercent = int.parse(stdin.readLineSync());
  taxPercent = int.parse(stdin.readLineSync());

  double tip = (meal_cost * tipPercent) / 100.0;
  double tax = (meal_cost * taxPercent) / 100.0;

  double output = (meal_cost + tip + tax).roundToDouble();
  print(output);
}
